docs/contributorfile.rst

```diff
@@ -132,75 +156,144 @@ Issues labels
+Other labels
+^^^^^^^^^^^^
+-  Invalid: These PRs don't seem right. They actually seem so not right that they won’t be further processed. This label invalidates a Hacktoberfest contribution. If you think this is wrong, start a discussion in the relevant issue (or open one if missing). Reviewers are asked to give an explanation for the use of this label.
 
 .. _g1i:
 
 Good First Issues
 ~~~~~~~~~~~~~~~~~
-Good First Issues are issues that are either very simple, or that help knowing the programs or the language better. We use it to help contributors with less experience to learn and familiarise with Git, GitHub, Python3, and physiology.
+Good First Issues are issues that are either very simple, or that help the contributor get to know the programs or the languages better. We use it to help contributors with less experience to learn and familiarise with Git, GitHub, Python3, and physiology.
 We invite more expert contributors to avoid those issues, leave them to beginners and possibly help them out in the resolution of the issue. However, if the issue is left unassigned or unattended for long, and it’s considered important or urgent, anyone can tackle it.
 
 .. _workflow:
 
 Contribution workflow
 ---------------------
 There are many descriptions of a good contribution workflow out there. For instance, we suggest to have a look at `tedana's workflow <https://github.com/ME-ICA/tedana/blob/master/CONTRIBUTING.md#making-a-change>`_.
-At physiopy, we follow a very similar workflow. The only two differences are:
+At ``physiopy``, we follow a very similar workflow. The only three differences are:
 
--  We ask you to test the code locally before merging it, and then, if possible, write some automatic tests for the code to be run in our Continuous Integration! Check the testing section below to know more.
--  We suggest opening a draft PR as soon as you can - so it’s easier for us to help you!
+- If you see an open issue that you would like to work on, check if it is assigned. If it is, ask the assignee if they need help or want to be substituted before starting to work on it.
+- We ask you to test the code locally before merging it, and then, if possible, write some automatic tests for the code to be run in our Continuous Integration! Check the testing section below to know more.
+- We suggest opening a draft PR as soon as you can - so it’s easier for us to help you!
 
 .. _pr:
 
 Pull Requests
 -------------
-To improve understanding pull requests "at a glance", we use the same labels used for issues. Multiple labels can be assigned - just think which ones suit your PR the most! 
-When opening a pull request, assign it to at least one label.
+To improve understanding pull requests "at a glance", we use the labels listed above. Multiple labels can be assigned, but it's a good idea to keep different types of contributions separated, unless they are minimal - for instance, you might want to open a PR for code, one for documentation and one for testing.
```

Comment:
This confuses me a little. It implies to me that one set of changes (e.g., adding a new metric to `phys2denoise`), would involve three PRs: one to add the metric, one to add documentation about the metric, and one to add tests for the metric. The last one clearly falls under the first PR based on the rest of this document, but I don't think it's clear from context if contributors should separate the PR with the new metric+testing from another PR with documentation about the metric. Personally, I think all of it should be done in one PR in that situation.

Reply:
The reasoning is two(three?)fold.
First, if we want to use the automatic `all-contributors` plugin for `auto`, if reviewers use the "suggestion" tool they are interpreted as co-authors of the PR. For instance, this PR is labelled "internal" and "documentation", so all reviewers I accepted suggestion from will get the contribution badge for `documentation` and `infrastructure`. Normally it doesn't matter so much because we tend to review the types of PR that match our previous contribution, but I already have to clean up a bit the contribution table.
Second, since the main focus of a `code` PR is code changes (and not tests or documentations, that have to be done but are _secondary_ in that context), even if the PR contains changes to the tests and documentation I would only label it with one of the two code-related label (majormod and minormod), since the code change is the real scope of the PR. This is also related to the fact that all-contributors has a binary nature (either you contributed in a field or you didn't). I find slightly unfair that two contributors have the same badge (e.g. `test`), but one made almost all the tests and the other two small changes. This is partially solved by the number and size of the PRs labelled in a certain way, so having clear, distinct PRs helps adding a metric of contribution.
Bonus reason: we could _not_ use the all-contributors plugin for `auto`, but most of the time it helps adding contributions impartially and it really helps keeping track of most of the contributions.

Reply:
Gotcha! Still, I think tests could go with code changes into the same PR. If we make two different PRs we will increase our version with each of them, which does not make sense to me if the PRs are 100% related like code + tests.

Reply:
I think it might make sense to include tests alongside the code, in the same PR, as well. 

And if a contributor writes the code, but is not sure about how to write the tests, that is when they can make a draft PR and ask for help from other people to include tests in their PR?

Reply:
I'm going to update after the discussion in the call!